components/webln.js

```diff
@@ -0,0 +1,59 @@
+import { createContext, useContext, useEffect, useState } from 'react'
+
+const WebLNContext = createContext({})
+
+export function WebLNProvider ({ children }) {
+  const [provider, setProvider] = useState(null)
+  const [info, setInfo] = useState(null)
+  // NOTE these functions are undefined initially - can this be a problem?
+  const [launchModal, setLaunchModal] = useState()
+  const [launchPaymentModal, setLaunchPaymentModal] = useState()
+  const [closeModal, setCloseModal] = useState()
+  const [isConnected, setIsConnected] = useState()
+
+  useEffect(() => {
+    const unsub = []
+    async function effect () {
+      const [isConnected, onConnected, onDisconnected, requestProvider, launchModal, launchPaymentModal, closeModal] = await import('@getalby/bitcoin-connect-react').then(
```

Comment:
you could import `init` here too and then call it to set Stacker News as the name of the connecting app (this will work for at least NWC and maybe other connectors in the future)